lib/service/servicecfg/config.go

```diff
@@ -579,6 +580,8 @@ func ApplyDefaults(cfg *Config) {
 // FileDescriptor is a file descriptor associated
 // with a listener
 type FileDescriptor struct {
+	once *sync.Once
```

Comment:
Nit: I figure this could have just been `sync.Once` and avoided the need for the constructor - but this is fine.